source/develop/release-management/features/storage/vm-leases.html.md

```diff
@@ -189,39 +182,72 @@ using a VM lease on another host.
 
 This verb is implemented since ovirt-3.5 as internal verb, we will make
 this API public in 4.1.
-See https://gerrit.ovirt.org/29157
+See [Gerrit patch 29157](https://gerrit.ovirt.org/29157).
 
 
 ### xleases volume operation
 
 
 #### volume format
 
-The first lease slot (offset 0) in the xleases volume is used for the
-index, mapping from lease id to lease offset.
+The xleases volume format was designed so future sanlock version
+implementing internal index can use the same format, and we can move
+lease management into sanlock.
+
+The volume is composed of slots, each one is 1MiB when the underlying
+storage is using 512 bytes sectors, or 8MiB for 4K sector size.
+Currently vdsm supports only 512 bytes sectors, but this part will
+support both now.
+
+The volume layout is:
+
+- Lockspace (1 slot) - used for delta leases. Vdsm will not use this
+  area since we are using the special "ids" volume. A future vdsm
+  version may switch to the xleases volume for the lockspace.
+
+- Index (1 slot) - used for mapping between lease id and the lease
+  offset.
+
+- Sanlock internal resource - used for protecting the leases volume so
+  it can be modified by any host. Vdsm will not use this resource now,
+  but we plan to use it in a future version so lease management is not
+  tied to the SPM.
+
+- User resources - VM leases are allocated in this area
+
+
+#### Leases volume index format
+
+The leases volume index is composed of blocks (512 or 4K bytes). The
+layout is:
+
+1. Metadata block
+2. Record blocks
+
+The metadata block holds metadata about the entire index.
+
+1. Magic number (similar to lockspace and resources slots)
+2. version
+3. lockspace name
+4. timestamp
+5. stale flag
```

Comment:
stale flag - meaning the entire index is stale. This will be used when formatting or rebuilding an index.